event/views.py

```diff
@@ -1,3 +1,77 @@
-from django.shortcuts import render
+from django.shortcuts import get_object_or_404
+from django.http import JsonResponse
+from django.contrib.auth.mixins import LoginRequiredMixin
+from rest_framework.views import APIView
+from rest_framework.permissions import IsAdminUser
+from .models import Event
+import json
 
-# Create your views here.
+class EventListView(APIView):
+    """
+    API view to retrieve a list of all events.
+    """
+    def get(self, request):
+        event = Event.objects.all()
+        event_list = list(event.values())
+        return JsonResponse({'event': event_list})
+
+class EventDetailView(APIView):
+    """
+    API view to retrieve details of a specific event.
+    """
+    def get(self, request, pk):
+        event = get_object_or_404(Event, pk=pk)
+        event_data = {
+            'title': event.title,
+            'description': event.description,
+            'image': event.image.url if event.image else None,  # Handle case where image is null
+            'date': event.date,
+            'location': event.location,
+        }
+        return JsonResponse({'event': event_data})
+
+class EventCreateView(LoginRequiredMixin, APIView):
+    """
+    API view to create a new event (requires login).
+    """
+    permission_classes = [IsAdminUser]
```

Comment:
This 'IsAdminUser' is referring to Django Default Admin User, and we won't be using that. There should be a field in the User model that signifies an Admin User, let's use that.